### PR TITLE
[Android] Fix Kotlin and Java deprecations

### PR DESCRIPTION
**Pull request by AenBleidd**
Fix deprecated ViewPager and NetworkInfor.
Update gradle.build to show Java deprecations.

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

**Review by talregev — CHANGES_REQUESTED** [review]
Please change by the following comments.

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/DeviceStatus.java [hunk 1]
```diff
@@ -213,6 +206,17 @@ private boolean isNetworkTypeWiFiOrEthernetOnAPI23AndHigher() {
         return false;
     }
 
+    @SuppressWarnings( "deprecation" )
+    private boolean isNetworkTypeWiFiOrEthernetOnAPI22AndLower() {
```
[the remark this answers is not in the document]
Comment (talregev) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/client/DeviceStatus.java:210]: Here recent example: 
https://github.com/BOINC/boinc/commit/ad4c091712fa144feef080d5faaef3218a021f95#diff-e759796c49aad28d27e13ed86fc5ff6ba470c68766e211f1e46266187e5f0751R127

File: android/BOINC/app/build.gradle [hunk 2]
```diff
@@ -143,6 +143,10 @@ android {
             }
         }
     }
+    tasks.withType(JavaCompile) {
+    options.compilerArgs << '-Xlint:unchecked'
+    options.deprecation = true
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply android/BOINC/app/build.gradle:148]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/BatchProcessingActivity.kt [hunk 3]
```diff
@@ -28,12 +28,15 @@ import android.os.IBinder
...
 import androidx.fragment.app.FragmentManager
-import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
 import androidx.lifecycle.lifecycleScope
-import androidx.viewpager.widget.ViewPager.OnPageChangeListener
+import androidx.viewpager2.widget.ViewPager2.*
 import edu.berkeley.boinc.BOINCActivity
 import edu.berkeley.boinc.R
+import edu.berkeley.boinc.attach.HintFragment.*
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/BatchProcessingActivity.kt:39]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/BatchProcessingActivity.kt [hunk 4]
```diff
@@ -28,12 +28,15 @@ import android.os.IBinder
...
 import android.view.View
 import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
-import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
 import androidx.lifecycle.lifecycleScope
-import androidx.viewpager.widget.ViewPager.OnPageChangeListener
+import androidx.viewpager2.widget.ViewPager2.*
```
[the remark this answers is not in the document]
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/BatchProcessingActivity.kt:36]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/DeviceStatus.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/client/DeviceStatus.java:210]: Fixed

**Review by talregev — APPROVED** [review]
I compile and run and crunch on OnePlus one (Android 10).
I approve

**Pull request merged**